pygeoapi/cql.py

```diff
@@ -0,0 +1,293 @@
+                return self.method.like(
+                    self.feature_list,
+                    to_filter(node.lhs),
+                    to_filter(node.rhs),
+                    node.case, node.not_
+                )
+
+            # evaluation for In Predicate Node
+            elif isinstance(node, InPredicateNode):
+                return self.method.contains(
+                    self.feature_list,
+                    to_filter(node.lhs),
+                    [to_filter(sub_node)
+                     for sub_node in node.sub_nodes],
+                    node.not_
+                )
+
+            # evaluation for Null Predicate Node
+            elif isinstance(node, NullPredicateNode):
+                return self.method.is_null(
+                    self.feature_list,
+                    to_filter(node.lhs),
+                    node.not_
+                )
+
+            # evaluation for Temporal Predicate Node
+            elif isinstance(node, TemporalPredicateNode):
+                return self.method.temporal(
+                    self.feature_list,
+                    to_filter(node.lhs),
+                    node.rhs,
+                    node.op
+                )
+
+            # evaluation for Spatial Predicate Node
+            elif isinstance(node, SpatialPredicateNode):
+                return self.method.spatial(
+                    self.feature_list,
+                    to_filter(node.lhs),
+                    to_filter(node.rhs),
+                    node.op,
+                    to_filter(node.pattern),
+                    to_filter(node.distance),
+                    to_filter(node.units)
+                )
+
+            # evaluation for BBox Predicate Node
+            elif isinstance(node, BBoxPredicateNode):
+                return self.method.bbox(
+                    self.feature_list,
+                    to_filter(node.lhs),
+                    to_filter(node.minx),
+                    to_filter(node.miny),
+                    to_filter(node.maxx),
+                    to_filter(node.maxy),
+                    to_filter(node.crs),
+                )
+
+            # evaluation for Attribute Expression Node
+            elif isinstance(node, AttributeExpression):
+                return self.method.attribute(node.name, self.field_list)
+
+            # evaluation for Literal Expression Node
+            elif isinstance(node, LiteralExpression):
+                return node.value
+
+            # return the Node
+            return node
+
+    class CQLFilter:
+        """ CQL Filter Executor """
+
+        def __init__(self):
+            """
+            Initialize object
+            """
+
+            self.CQLParser = self.CQLParser
+            self.cql_expression = self.cql_expression
+            self.CQLEvaluator = self.CQLEvaluator
+            self.feature_list = self.feature_list
+            self.field_list = self.field_list
+            self.CQLFilter = self.CQLFilter
+
```

Comment:
First of all: great and very useful contribution! Trying to understand the use and lifecycle of Inner Classes here: 

* looks like `CQLFilter` is never instantiated (?) so the `CQLFilter.__init__(self)` is obsolete(?)
* lines 228-233 seem to do nothing, or is there some clever data passing from Outer (FilterHandler) to Inner Class?
* methods of `CQLFilter` are basically called as static/factory methods from `CQLHandler`. See e.g. line 44: `feature_list = self.CQLFilter.get_cql_filtered_list(self)`. Is in e.g. `get_cql_filtered_list(self)` `self` then an instance to the Outer FilterHandler class?
* maybe make the use of static functions in `CQLFilter` more explicit with @static and arguments denoting the FilterHandler object i.s.o. `self`.